### DISCUSSION TITLE
Keep key_table when using set_config_overrides?

### DISCUSSION THREAD

**Discussion by hereGittyGitty** [wezterm/wezterm#3230] [Q&A] [answered]
Hey there,

I am trying to scroll through _color schemes_ by activating a `key_table` with `one_shot = false` and iteratively setting the `color_scheme` variable by emitting events when pressing the keys of that `key_table`. Right now, it seems like calling `set_config_overrides` automatically _pops_ the `key_table` from the stack.

My minimal configuration to show that behavior looks like this:

```lua
local wezterm = require("wezterm")
local act = wezterm.action

wezterm.on("update-right-status", function(window, pane)
  local name = window:active_key_table()
  if name then
    name = "TABLE: " .. name
  end
  window:set_right_status(name or "")
end)

wezterm.on("trigger-good-morning", function(window, pane)
  wezterm.log_info("good morning")
  window:set_config_overrides({
    color_scheme = "3024 Day",
  })
end)

wezterm.on("trigger-good-night", function(window, pane)
  wezterm.log_info("good night")
  window:set_config_overrides({
    color_scheme = "3024 Night",
  })
end)

return {
  use_dead_keys = false,
  leader = { key = "Space", mods = "CTRL|SHIFT" },

  keys = {
    {
      key = "t",
      mods = "LEADER",
      action = act.ActivateKeyTable({ name = "check_time", one_shot = false }),
    },
  },

  key_tables = {
    check_time = {
      { key = "m", action = act.EmitEvent("trigger-good-morning") },
      { key = "n", action = act.EmitEvent("trigger-good-night") },

      -- Cancel the mode by pressing escape
      { key = "Escape", action = "PopKeyTable" },
    },
  },
}
```
Is it possible, to keep the `key_table` when setting a _color scheme_?

Thanks :)

**Comment by wez** [comment, marked as answer]
The key table stack is completely cleared when the configuration is changed; it's a feature intended to provide a way to break out of situations where you may have produced key tables that render you unable to to send input to the terminal.  Updating the config will return you to the ground state and restore your ability to interact with the terminal.

Setting config overrides is a variation of changing the configuration, so it also triggers this behavior.

There isn't a way to prevent this behavior.
